filters/fadein/fadein.go

```diff
@@ -0,0 +1,252 @@
+package fadein
+
+import (
+	"fmt"
+	"net"
+	"net/url"
+	"strings"
+	"time"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/zalando/skipper/eskip"
+	"github.com/zalando/skipper/filters"
+	"github.com/zalando/skipper/routing"
+)
+
+const (
+	DurationName        = "fadeInDuration"
+	EaseName            = "fadeInEase"
+	EndpointCreatedName = "endpointCreated"
+)
+
+type (
+	duration time.Duration
+	ease     float64
+
+	endpointCreated struct {
+		when  time.Time
```

Comment:
"when": what does it mean? Creation time?

Reply:
yes. The 'created' term is already noted in the name of the type, I thought it's simpler to call it this way. It also could be 'value', maybe that's the best.

Reply:
See similar, though: https://github.com/golang/go/blob/master/src/time/sleep.go#L15

Reply:
I see ok makes sense "detected fade in -> when", but maybe `start` sounds more good?
Not sure, I will think about this while working on the code

Reply:
Yeah this "when" makes sense.